include/emds/emds.h

```diff
@@ -0,0 +1,189 @@
+ *            value.
+ * @param _data Data pointer to be stored at emergency data store.
+ * @param _len Length of data to be stored at emergency data store.
+ *
+ * This creates a variable _name prepended by emds_.
+ */
+#define EMDS_STATIC_ENTRY_DEFINE(_name, _id, _data, _len)                      \
+	static const STRUCT_SECTION_ITERABLE(emds_entry, emds_##_name) = {     \
+		.id = _id,                                                     \
+		.data = (uint8_t *)_data,                                      \
+		.len = _len,                                                   \
+	}
+
+/**
+ * @brief Initialize the emergency data storage.
+ *
+ * Initializes the emergency data storage. This need to called before starting
+ * to add entries to the emds_dynamic_entry and doing restore.
+ *
+ * @retval 0 Success
+ * @retval -ERRNO errno code if error
+ */
+int emds_init(void);
+
+/**
+ *  @brief Add entry that should be saved/restored when emergency data storage is called.
+ *
+ * This will add the entry to the dynamic entry list, and will when the
+ * @ref 'emds_store' function is called take the given data pointer and store
+ * the data to the emergency data storage.
+ *
+ * @param entry Entry to add to list.
+ *
+ * @retval 0 Success
+ * @retval -ERRNO errno code if error
+ */
+int emds_entry_add(struct emds_dynamic_entry *entry);
+
+/**
+ * @brief Start the emergency data storage process.
+ *
+ * This will trigger the process of storing all data that has been regitster to
+ * be stored. When it is finished with storing the data the data should not be
+ * changed and the board should be halted or rebooted at the end. This is a time
+ * critical process and will be processed as fast as possible. This will store
+ * all data register either though the 'emds_entry_add' or 'EMDS_ENTRY_DEFINE'.
+ *
+ * @retval 0 Success
+ * @retval -ERRNO errno code if error
+ */
+int emds_store(void);
+
+/**
+ * @brief Load all data from the emergency data storage.
+ *
+ * This need to be called after the static and dynamic entries are added, as
+ * they are used to select data that should be loaded. This should also be
+ * called before the 'emds_prepare' function is called as that will delete all
+ * the stored data.
+ *
+ * @retval 0 Success
+ * @retval -ERRNO errno code if error
+ */
+int emds_load(void);
+
+/**
+ * @brief Clear flash area from the emergency data storage.
+ *
+ * This function will clear the flash area for all previous stored data.
+ *
+ * @retval 0 Success
+ * @retval -ERRNO errno code if error
+ */
+int emds_clear(void);
+
+/**
+ * @brief Prepare flash area for a next emergency data storage.
+ *
+ * This function will prepare the flash area for the next emergency data
+ * storage. This delete the current entries if it enough space for the next
+ * emergency data storage. It will clear the flash storage if it is not enough
+ * space for the next storage. This has to be done after all the dynamic entries
+ * are added.
```

Comment:
```suggestion
 * @brief Prepare flash area for the next emergency data storage.
 *
 * This function prepares the flash area for the next emergency data
 * storage. It deletes the current entries if there is enough space for the next
 * emergency data storage, and clears the flash storage if there is not enough
 * space for the next storage. This has to be done after all the dynamic entries
 * are added. 
```
Check if I understood this correctly, and did not change the meaning.